packages/tasit-action/src/Contract.test.js

```diff
@@ -103,73 +103,162 @@ describe("TasitAction.Contract", () => {
-    const value = await simpleStorage.getValue();
+    const value = await sampleContract.getValue();
     expect(value).to.exist;
   });
 
   describe("wallet/account setup tests", async () => {
     it("should throw error when setting a wallet with no wallet argument", async () => {
       expect(() => {
-        simpleStorage.setWallet();
+        sampleContract.setWallet();
       }).to.throw();
     });
 
     it("should throw error when setting invalid wallet", async () => {
       expect(() => {
-        simpleStorage.setWallet("invalid wallet");
+        sampleContract.setWallet("invalid wallet");
       }).to.throw();
     });
 
     it("should throw error when calling write method without account/wallet", async () => {
       expect(() => {
-        simpleStorage.setValue("hello world");
+        sampleContract.setValue("hello world");
       }).to.throw();
     });
 
     it("should throw error when calling write method after account/wallet removal", async () => {
-      simpleStorage.setWallet(wallet);
-      simpleStorage.removeWallet();
+      sampleContract.setWallet(wallet);
+      sampleContract.removeWallet();
       expect(() => {
-        simpleStorage.setValue("hello world");
+        sampleContract.setValue("hello world");
       }).to.throw();
     });
   });
 
-  describe("TransactionSubscription - actions (tx) subscriptions tests", async () => {
+  describe("Contract Subscription", async () => {
+    beforeEach("assign a wallet to the contract", () => {
+      expect(() => {
+        sampleContract.setWallet(wallet);
+      }).not.to.throw();
+    });
+
+    it("should trigger Contract error event on Action error", async () => {
+      const errorListener = sinon.fake();
+
+      sampleContract.on("error", errorListener);
+
+      txSubscription = sampleContract.revertWrite("some string");
+      await txSubscription.waitForNonceToUpdate();
+
+      expect(errorListener.callCount).to.equal(1);
+    });
+
+    it("should trigger Contract and TransactionSubscription error events on Action error", async () => {
+      const contractErrorListener = sinon.fake();
+      const txErrorListener = sinon.fake();
+
+      sampleContract.on("error", contractErrorListener);
+
+      txSubscription = sampleContract.revertWrite("some string");
+
+      txSubscription.on("error", txErrorListener);
+
+      await txSubscription.waitForNonceToUpdate();
+
+      expect(contractErrorListener.callCount).to.equal(1);
+      expect(txErrorListener.callCount).to.equal(1);
+    });
+
+    it("should trigger Contract error event on Event error", async () => {
+      const errorListener = sinon.fake();
+      const eventFnFake = sinon.fake();
+      const eventListener = () => {
+        eventFnFake();
+        throw new Error();
+      };
+
+      sampleContract.on("error", errorListener);
+
+      contractSubscription = sampleContract.subscribe();
+
+      // Tech debt: Know why changing to .once() eventListener stills be called twice
```

Comment:
```suggestion
      // Tech debt: Know why changing to .once() eventListener still is called twice
```

Reply:
Deprecated. The comment was deleted. I'm resolving this.